### DISCUSSION TITLE
SFTP server directory listing fails with curl

### DISCUSSION THREAD

**Discussion by umeshauv** [drakkan/sftpgo#1161] [Q&A] [answered]
Hi
  I have configured SFTP on CentOS 7 server and I am able to login using username and password and able to see directories and files inside individual directories.   However, when I use curl like curl -u user1:<password> sftp://user1@<sftp_host>:<sftp_port>/dir directory listing fails with error "curl: (56) Failure when receiving data from the peer".   

![Screen Shot 2023-01-23 at 2 22 36 PM](https://user-images.githubusercontent.com/70544075/214130715-d23f8f7c-58b2-4b38-905f-f0d2b35964ce.png)

![Screen Shot 2023-01-23 at 2 16 18 PM](https://user-images.githubusercontent.com/70544075/214129679-4caa6815-99b3-41d3-bc8b-d283731a7195.png)

I can list the directory, but I cannot list the contents of the directory which is causing an issue.
curl -u user1:<password> -k sftp://user1@10.0.8.4:8822/
drwxr-xr-x    1 0        0               0 Jan 1  1970 .
drwxr-xr-x    2 996      992            54 Jan 23 19:21 test_data
drwxr-xr-x    1 0        0               0 Jan 1  1970 global_ref_data

$ curl -u user1:<password> -k sftp://user1@10.0.8.4:8822/global_ref_data
curl: (56) Failure when receiving data from the peer
$ curl -u user1:<password> -k sftp://user1@10.0.8.4:8822/test_data
curl: (56) Failure when receiving data from the peer

Appreciate your help.

Thanks,
Umesh

**Comment by drakkan** [comment, marked as answer]
end the URL with a trailing slash to indicate to curl that it is a directory

**Reply by umeshauv** [reply]
Thanks. It worked.
